fre/pp/configure_script_xml.py

```diff
@@ -174,67 +193,93 @@ def main(xml, platform, target, experiment, do_analysis, historydir, refinedir,
-                                                                        experiment)
-    logging.info(">> {}".format(fetch_analysis_dir_cmd))
-    fetch_analysis_dir_process = subprocess.run(fetch_analysis_dir_cmd,
-                                      shell=True,
-                                      check=True,
-                                      capture_output=True,
-                                      universal_newlines=True)
-    analysisDir = fetch_analysis_dir_process.stdout.strip()
-    logging.info(analysisDir)
+    if historydir is None:
+        logging.info("Running frelist for historydir assignment, this may fail")
+        logging.info("If so, try the frelist call manually and use the historydir argument")
+
+        fetch_history_cmd = f"frelist -x {xml} -p {platform} -t {target} {experiment} -d archive"
+        logging.info(">> %s", fetch_history_cmd)
+
+        fetch_history_process = subprocess.run(fetch_history_cmd,
+                                               shell=True,
+                                               check=True,
+                                               capture_output=True,
+                                               universal_newlines=True)
+        historydir = fetch_history_process.stdout.strip() + '/history'
+        logging.info(historydir)
 
-    gridSpec = frelist_xpath(xml, platform, target, experiment, do_analysis, historydir, refinedir, ppdir, do_refinediag, pp_start, pp_stop, validate, verbose, quiet, dual, 'input/dataFile[@label="gridSpec"]')
-    simTime = frelist_xpath(xml, platform, target, experiment, do_analysis, historydir, refinedir, ppdir, do_refinediag, pp_start, pp_stop, validate, verbose, quiet, dual, 'runtime/production/@simTime')
-    simUnits = frelist_xpath(xml, platform, target, experiment, do_analysis, historydir, refinedir, ppdir, do_refinediag, pp_start, pp_stop, validate, verbose, quiet, dual, 'runtime/production/@units')
+
+
+    # Q: should respond to do_refineDiag and refinedir args of this function?
+    historydir_refined = historydir + '_refineDiag'
+
+    if ppdir is None:
+        logging.info("Running frelist for ppdir assignment...")
+        logging.info("If this fails, try the frelist call manually and use the ppdir argument")
+        fetch_pp_cmd = f"frelist -x {xml} -p {platform} -t {target} {experiment} -d postProcess"
+        logging.info(">> %s", fetch_pp_cmd)
+
+        fetch_pp_process = subprocess.run(fetch_pp_cmd,
+                                          shell=True,
+                                          check=True,
+                                          capture_output=True,
+                                          universal_newlines=True)
+        ppdir = fetch_pp_process.stdout.strip()
+        logging.info(ppdir)
+
+
+    # Q: shouldn't there be a CLI analysis dir arg while we're here?
+    # basically, this is borderline on the same level as the ppDir and historydir fields.
```

Comment:
It's painful to have to retrieve them separately with separate frelist calls, is that what you're saying?

Reply:
more like, since `ANALYSIS_DIR` can be set by hand in a config file and everything works out just fine, might as well make it an input arg IMO. 

sometimes, being able to make I/O directories explicit is nice